script.js

```diff
@@ -1,4 +1,235 @@
+var PLAYER = 'player'
+var COMPUTER = 'computer'
+var currentGameMode = GAME_MODE_DEAL_CARDS; //initialise dealing of cards
+var currentPlayer = 'player'
+var gameOver = false;
+
+//=======================Create 52-card deck=====================
+var makeDeck = function () {
+  // create the empty deck at the beginning
+  var deck = [];
+  var suits = ['hearts', 'diamonds', 'clubs', 'spades'];
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    // make a variable of the current suit
+    var currentSuit = suits[suitIndex];
+    // loop to create all cards in this suit
+    // rank 1-13
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+
+      // 1, 11, 12 ,13
+      if (cardName == 1) {
+        cardName = 'ace';
+      } else if (cardName == 11) {
+        cardName = 'jack';
+      } else if (cardName == 12) {
+        cardName = 'queen';
+      } else if (cardName == 13) {
+        cardName = 'king';
+      }
+
+      // make a single card object variable
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // add the card to the deck
+      deck.push(card);
+      rankCounter = rankCounter + 1;
+      console.log(rankCounter)
+    }
+    suitIndex = suitIndex + 1;
+    console.log(suitIndex)
+  }
+  return deck;
 };
+
+var getRandomIndex = function (size) {
+  return Math.floor(Math.random() * size);
+};
+
+var shuffleCards = function (cards) {
+  var index = 0;
+
+  while (index < cards.length) {
+    var randomIndex = getRandomIndex(cards.length);
+    var currentItem = cards[index];
+    var randomItem = cards[randomIndex];
+    cards[index] = randomItem;
+    cards[randomIndex] = currentItem;
+    index = index + 1;
+  }
+
+  console.log(cards)
+  return cards;
+};
+//=======================Shuffle cards once=====================
+var deck = makeDeck();
+var shuffledDeck = shuffleCards(deck);
+
+// Return a string of ranks and suits of cards in the hand array
+var printCards = function (hand) {
+  var returnString = '';
+  // Iterate until hand.length - 1 so we can avoid the extra comma at the end of return string
+  for (var i = 0; i < hand.length - 1; i += 1) {
+    var newCard = `${hand[i].name} of ${hand[i].suit}`
+    returnString = returnString + ',' + newCard;
+  }
+  return returnString;
+};
+
+//deal cards to hands
+var dealCardToHand = function () {
+  for (var i = 0; i < 2; i ++) {
+    hand.push(shuffledDeck.pop())
+  }
+  return hand;
+};
+
+//=======================Store player and computer's cards=====================
+var playerHand = [];
+var computerHand = [];
+let maxCardLimit = 21; //max value before bust is 21
+let dealerMinValue = 16; // dealer has to minimally deal 16
+let numOfCards = 0;
+//=======================Main function=====================
+var main = function (input){
+  var myOutputValue = '';
+  currentGameMode = GAME_MODE_DEAL_CARDS
```

Comment:
if you put this line inside the main function, the `currentGameMode` will always be `GAME_MODE_DEAL_CARDS` when the main function runs. Try to extract line 113 to 128 into a helper function and name it something like `conductInitialDealingRound`